Toplean_Alexandru/week-02/Domain/Domain/DeleteRestaurantOp/DeleteRestaurantOp.cs

```diff
@@ -16,12 +16,17 @@ public override Task<IDeleteRestaurantResult> Work(DeleteRestaurantCmd Op, Unit
         {
             if (Exists(Op.RestaurantName))
             {
+                // Get the restaurant that will be deleted
                 Restaurant restaurant = AllHardcodedValues.HarcodedVals.RestaurantList.FirstOrDefault(r => r.Name == Op.RestaurantName);
+                // Deleting the restaurant
                 AllHardcodedValues.HarcodedVals.RestaurantList.Remove(restaurant);
```

Comment:
What happens if the restaurant doesn't exist? FirstOrDefault will return null, which will most likely blow up your code. Should you adapt your response to this case?